Makefile

```diff
@@ -1,25 +1,125 @@
-.PHONY: build up backend
+.PHONY: up down clean clean_volumes clean_orphans clean_images clean_all re build_all up_nginx up_backend up_db up_elk up_all up_no_elk logs_nginx logs_backend logs_db logs_elk status exec_nginx exec_backend exec_db exec_elk stop_all help
 
-# Create directories for mounting volumes
+# Default target: Build and start all services except ELK stack
+all: up_no_elk
 
-all: build up
+# Build all Docker images
+build_all:
+	docker-compose -f ./docker-compose.yml build
 
-# Build the Docker images
-build: 
-	docker compose -f docker-compose.yml build
+# Start all services except ELK stack and rebuild if necessary
+up_no_elk:
+	docker-compose -f ./docker-compose.yml up --build -d nginx backend_dummy db
 
-# Start the services
-up: build	
-	docker compose -f docker-compose.yml up -d
+# Start all services including ELK stack and rebuild if necessary
+up_all:
+	docker-compose -f ./docker-compose.yml up --build -d
 
-# Rebuild and restart the backend service
-backend:
-	docker compose -f docker-compose.yml build backend_dummy
-	docker compose -f docker-compose.yml up -d backend_dummy
+# Start specific services and rebuild if necessary
+up_nginx:
+	docker-compose -f ./docker-compose.yml up --build -d nginx
 
-# Stop the services
-clean:
-	docker rm -vf $$(docker ps -aq)
+up_backend:
+	docker-compose -f ./docker-compose.yml up --build -d backend_dummy
+
+up_db:
+	docker-compose -f ./docker-compose.yml up --build -d db
+
+up_elk:
+	docker-compose -f ./docker-compose.yml up --build -d es01 kibana logstash01 filebeat01 metricbeat01
+
+# Stop and remove containers, networks, and volumes
+down:
+	docker-compose -f ./docker-compose.yml down
+
+# Clean up volumes
+# Use this to remove volumes and reset persistent data
+clean_volumes:
+	docker-compose -f ./docker-compose.yml down -v
+
+# Clean up orphans
+# Use this to remove containers that are no longer defined in the current docker-compose.yml
+clean_orphans:
+	docker-compose -f ./docker-compose.yml down --remove-orphans
+
+# Clean up images
+# Use this to remove all Docker images
+clean_images:
 	docker rmi -f $$(docker images -aq)
 
-re: clean all
\ No newline at end of file
+# Clean up all: containers, networks, volumes, and images
+# Use this for a full cleanup of the Docker environment
+clean_all: clean_volumes clean_orphans clean_images
+
+# Clean up: containers, networks, volumes, and orphans (default clean)
+# Use this to reset data and ensure only defined services are running
+clean: clean_volumes clean_orphans
+
+# Rebuild and start all services
+re: clean_all up_no_elk
```

Comment:
it needs to be modified make re, because now you delete the volumes but it should not be delete it. por favor trabaja bien !

Reply:
This issue has been solved.